certificate-manager/src/main/java/io/strimzi/certs/OpenSslCertManager.java

```diff
@@ -216,139 +126,508 @@ private File buildConfigFile(Subject sbj, boolean isCa) throws IOException {
+                    .optArg("-keyfile", tmpKey)
+                    .optArg("-startdate", notBefore)
+                    .optArg("-enddate", notAfter)
+                    .optArg("-subj", subject)
+                    .optArg("-config", defaultConfig)
+                    .database(database)
+                    .newCertsDir(newCertsDir)
+                    .basicConstraints("critical,CA:true,pathlen:" + pathLength)
+                    .keyUsage("critical,keyCertSign,cRLSign")
+                    .exec();
+
+            // The key will be in pkcs#1 format (bracketed by BEGIN/END RSA PRIVATE KEY)
+            // Convert it to pkcs#8 format (bracketed by BEGIN/END PRIVATE KEY)
+            new OpensslArgs("openssl", "pkcs8")
+                    .opt("-topk8").opt("-nocrypt")
+                    .optArg("-in", tmpKey)
+                    .optArg("-out", keyFile)
+                    .exec();
+        } finally {
+            delete(tmpKey);
+            delete(database);
+            delete(newCertsDir);
+            delete(csrFile);
+            delete(defaultConfig);
+        }
+    }
 
-            if (sbj.subjectAltNames() != null && sbj.subjectAltNames().size() > 0) {
+    public void generateIntermediateCaCert(File issuerCaKeyFile, File issuerCaCertFile,
+                                           Subject subject,
+                                           File subjectKeyFile, File subjectCertFile,
+                                           ZonedDateTime notBefore, ZonedDateTime notAfter, int pathLength) throws IOException {
+        // Preconditions
+        Objects.requireNonNull(issuerCaKeyFile);
+        Objects.requireNonNull(issuerCaCertFile);
+        Objects.requireNonNull(subject);
+        Objects.requireNonNull(subjectKeyFile);
+        Objects.requireNonNull(subjectCertFile);
+        checkValidity(notBefore, notAfter);
+        if (pathLength < 0) {
+            throw new IllegalArgumentException("pathLength cannot be negative: " + pathLength);
+        }
+        if (subject.subjectAltNames() != null && !subject.subjectAltNames().isEmpty()) {
+            throw new IllegalArgumentException("CA certificates should not have Subject Alternative Names");
+        }
 
-                // subject alt names need to be in an openssl configuration file
-                sna = buildConfigFile(sbj, false);
-                cmd.addAll(asList("-config", sna.toPath().toString(), "-extensions", "v3_req"));
-            }
+        // Generate a CSR for the key
+        Path tmpKey = null;
+        Path defaultConfig = null;
+        Path csrFile = null;
+        Path newCertsDir = null;
+        Path database = null;
 
-            cmd.addAll(asList("-subj", sbj.toString()));
+        try {
+            tmpKey = Files.createTempFile(null, null);
+            // Generate a key pair
+            new OpensslArgs("openssl", "genrsa")
+                    .optArg("-out", tmpKey)
+                    .opt("4096")
```

Comment:
because it's used in a couple of places, maybe it would make sense to use a key lenght constant?